agroal-narayana/src/main/java/io/agroal/narayana/NarayanaTransactionIntegration.java

```diff
@@ -103,12 +107,26 @@ public void associate(TransactionAware transactionAware, XAResource xaResource)
 
     @Override
     public boolean disassociate(TransactionAware connection) throws SQLException {
-        if ( transactionRunning() ) {
+        if ( transactionActive() ) {
             transactionSynchronizationRegistry.putResource( key, null );
         }
         return true;
     }
 
+    private boolean transactionActive() throws SQLException {
```

Comment:
So you have added this but kept the method `transactionRunning()`: what is the relationship between the two methods?

Reply:
TBH, I have no idea. `transactionRunning` is only used here
https://github.com/agroal/agroal/blob/697da5a3d56c4a08a8e8e77217fbb0e99962b76e/agroal-pool/src/main/java/io/agroal/pool/ConnectionHandler.java#L393-L400

Reply:
Also your new method is replacing the method `transactionRunning` which contains:

```
            if ( transaction == null ) {
                // AG-183 - Report running transaction when reaper thread attempts rollback
                return TxUtils.isTransactionManagerTimeoutThread();
            }
```
but in your replacement you just return false if ( transaction == null ). Why don't you call the same method when it is null?

Reply:
> So if you change that reference then the method verifyEnlistment will invoke your new method.

> Was it intentional?

Yeah, `verifyEnlistment` is used in `transactionCommit` and `transactionRollback` which I think they are related to `LocalXAResource`. As far as I read the codes, it looks like a `XAResourceWrapper`  to let a local tranaction to participant in a 2PC transaction.

If we reference the `transactionActiveCheck` to the new `transactionActive` method, `verifyEnlistment` will throw an Exception with  `Enlisted connection used without active transaction` when doing `transactionCommit` or `transactionRollback` because narayana thinks  that a  transaction in `COMMITTING` phase is **NOT** active.

Reply:
I think we need some guys from agroal to clarify what is the `transactionRunning` intention?

Reply:
@barreiro Please can you take a look at this PR?

Reply:
@zhfeng @mmusgrov the intention of `transactionRunning` is to prevent the connection to cross a transaction boundary (it can either be acquired within the scope or outside). For that purpose, the transaction is considered to be running during the commit/rollback phases to prevent commit of work outside the tx.

Your definition of `transctionActive` is different, and essentially means that the tx is 'usable' for client code.

My question is how do both concepts co-exist. For instance, does acquiring a connection during COMMITING phase means that it's life-cycle _should_ or _shouldn't_ be tied to the running transaction ??

Reply:
Thanks @barreiro !

It is a good question. So @mmusgrov in this case, from narayana side the following methods are allowed in `transactionSynchronizationRegistry` during `COMMITING` phase? is there anything in JTA spec?

- `getResource( key )` 
- `putResource( key )`
- `registerInterposedSynchronization(Synchronization synchronization)`

I think when acpuiring a connection, it needs to get a `TransactionAware` from the current transaction?

Reply:
For `getResource` and `putResource` there has to be transaction associated with the calling thread and it needs to be active (`Status.STATUS_ACTIVE`) and not marked rollback only (`Status.STATUS_MARKED_ROLLBACK`). During the commit phase the spec says that afterCompletion callbacks will be invoked in an undefined context.

Interposed synchronizations cannot be added after the transaction has prepared.